src/lib/comms/sol-netctl-impl-connman.c

```diff
@@ -328,8 +487,317 @@ get_service_ip(sd_bus_message *m,
     } while (1);
 
 end:
-    if (r == 0)
+    exit = sd_bus_message_exit_container(m);
```

Comment:
you don't need another variable here `exit`. You can re-use `r`, the return result using exit or  r here will be the same.